lazy_tensor_core/lazy_tensor_core/csrc/ts_backend/ts_node_lowering.cpp

```diff
@@ -174,12 +180,30 @@ class TSNodeLowering : public NodeLowering {
       case at::aten::ne: {
         return InferComparison(node);
       }
+      case at::aten::mean: {
+        auto mean = ir::NodeCast<ir::ops::Mean>(
+            node, ir::OpKind(at::aten::mean));
+        const ir::Output& argument = node->operand(0);
+        const lazy_tensors::Shape& argument_shape = argument.shape();
+        lazy_tensors::PrimitiveType element_type =
+        mean->dtype_ ? torch_lazy_tensors::TensorTypeToLtcType(*mean->dtype_)
+                     : argument_shape.element_type();
+        return lazy_tensors::Shape(element_type, argument_shape.dimensions());
+      }
       default:
         LTC_LOG(FATAL) << *node << "Not implemented yet.";
     }
   }
 
+
   TSOpVector LowerToTS(const ir::Node* node) {
+    // Desirable to have a way to do some ops via codegen and others by hand, 
+    // at least for now
+    auto codegen_lowering = LowerToTSCodegen(function_, loctx(), node);
+    if (codegen_lowering.size() > 0){
```

Comment:
Just wondering what your thoughts are - I can also imagine a version of this where each node class has its own (virtual) shape inference function, which would get rid of the giant switch statement

Reply:
Yep, I think that might be a good way to go.  XLA currently has it that way and Alex separated it out.  I think i'll leave that change as a separate PR if you're OK with it, since it probably makes sense to look holistically at the design of node class  rather than making the change reflexively

Reply:
I think the main issue with having 'Lower' as a function on Node base is that Lower would return a backend-specific object.  We could rethink its API so it returns a general LowerResult class that holds an opaque reference to a real backend lower result.

Reply:
Yeah - I guess right now, the two plugins still have some pretty large differences: the API each plugin needs to implement the `NodeLowering` interface, and xla and ts use two totally different classes under the hood to do it: `XlaOpVector and TSOpVector`. I could imagine that it'd be useful to try to figure out what the common interface between those types is so more of the logic could be implemented directly in LazyTensor instead of getting pushed to the plugins. But I bet that'll take a fair amount of design effort, and it seems like we can kind of paper over that for now with more codegen (aka what this PR currently does).

Reply:
Also, totally not in scope for this PR but something I'm just trying to understand better: I see that this PR still has a second giant switch statement (which is *not* codegen'd) at the `Infer` function here:  https://github.com/pytorch/pytorch/pull/64104/files#diff-64c166802d4c26f15e7fafd4efe9f2c19def21142d732db9aa2b7b300eaa84fdR79. I'm still trying to figure out - is the `Infer` function orthogonal to the codegen'd shape inference that you're generating in this PR? Or can any of it be re-used (or can the Infer function even be killed in favor of the shape inference you're doing). I have a feeling this has to do a least partially with the difference between xla vs. aten shape inference, but I'll need to stare at it some more.

Reply:
The link above didn't seem to be to an Infer function but I think I know what you're talking about.  I didn't want to change too many things at once.  We'll clean up the design of shape inference more holistically.  

One reason I didn't do more now is that I didn't want to have to codegen all the ops yet.  We still have ops which have manual TS lowerings, and for those we also have the manual TS shape inference pathway.  Once we move to codegen we'll naturally delete that code, and it'll move us toward the clean design.

On the XLA side, it's not clear yet if they will totally adopt our meta-inferred shapes, or if they will use it only for hashing but use their own lowering code to compute 'real' dtypes for example, and use those for some other parts.